src/logger/lwt_log_core.ml

```diff
@@ -47,6 +47,17 @@ let string_of_level = function
   | Error -> "error"
   | Fatal -> "fatal"
 
+let level_of_string str =
+  let str = (String.lowercase [@ocaml.warning "-3"]) str in
```

Comment:
You're right to worry about this, but we can't use [`lowercase_ascii`](http://caml.inria.fr/pub/docs/manual-ocaml/libref/String.html#VALlowercase_ascii) because Lwt still supports 4.02.